ui/src/cpu_inspector.rs

```diff
@@ -42,11 +49,24 @@ impl View for CpuInspector {
             }
             ui.text_edit_singleline(&mut cartridge_name);
             if ui.button("▶").clicked() {
-                // Start a thread for gameboy execution
-                todo!("Start a thread for a background gameboy execution");
+                if self.thread_handle.is_some() {
+                    return;
+                }
+
+                let gba_clone = Arc::clone(&self.gba);
+                let play_clone = Arc::clone(&self.play);
+
+                self.play.swap(true, std::sync::atomic::Ordering::Relaxed);
+
+                self.thread_handle = Some(thread::spawn(move || {
+                    while play_clone.load(std::sync::atomic::Ordering::Relaxed) {
+                        gba_clone.lock().unwrap().cpu.step();
+                    }
+                }));
```

Comment:
Not sure about the semantic of this... The emulator work like
1. exec n steps
2. receives interrupt after MHz converted to number of step
3. refresh with the current value of "things"
4. restart

I didn't try the branch but maybe this will be out of sync with the GUI
A sketched code can be
```
loop {
   while cpu.need_irq() {
        cpu.step()
   }

   refresh_gui();
   update_other_thing();
}
```

Reply:
Mhh I see what you mean but for now we don’t have Irqs and we don’t even know how many cycles take each instruction